doc/contributing/writing-tests.md

```diff
@@ -141,6 +141,26 @@ request. Interesting things to notice:
 
 ## General recommendations
 
+### Usage of `node:test`
+
+It is recommended to use `node:test` in tests outside of testing the `node:test`
+module, if particular functionality that's tested is not a dependency of
+`node:test` module. This ensures that a bug in the test runner doesn't impact
+the outcome of the underlying dependencies' test results.
+
+These dependencies are:
+
+- `node:async_hooks`
+- `node:child_process`
+- `node:fs`
+- ReadableStream in `node:streams`
+- `node:vm`
```

Comment:
I think the files listed in `test/parallel/test-bootstrap-modules.js` can be a good measure here.

Reply:
It's probably worth adding anything related to the bootstrapping process to the list of things not to test with the test runner since I'm not sure you can be 100% certain the test runner itself is bootstrapped properly at that point.

Reply:
(Most of) the files listed there are essential parts of the Node.js functionality that are used more ubiquitously, hence more likely to be depended on by `node:test` itself (e.g. when we talk about `node:async_hooks`, that's actually built on top of other modules, not just itself, `test/parallel/test-bootstrap-modules.js` list a set of files that are generally used everywhere)

Reply:
Can you recommend changes to the text, please?